src/seedu/addressbook/data/person/ReadOnlyPerson.java

```diff
@@ -66,20 +70,36 @@ default String getAsTextShowAll() {
      */
     default String getAsTextHidePrivate() {
         final StringBuilder builder = new StringBuilder();
-        builder.append(getName());
+        Printable[] printablesToPrint = new Printable[4];
+        printablesToPrint[NAME_INDEX] = getName();
         if (!getPhone().isPrivate()) {
-            builder.append(" Phone: ").append(getPhone());
+        	printablesToPrint[PHONE_INDEX] = getPhone();
         }
         if (!getEmail().isPrivate()) {
-            builder.append(" Email: ").append(getEmail());
+        	printablesToPrint[EMAIL_INDEX] = getEmail();
         }
         if (!getAddress().isPrivate()) {
-            builder.append(" Address: ").append(getAddress());
+        	printablesToPrint[ADDRESS_INDEX] = getAddress();
         }
-        builder.append(" Tags: ");
+        builder.append(getPrintableString(printablesToPrint));
+        builder.append("Tags: ");
         for (Tag tag : getTags()) {
             builder.append(tag);
         }
         return builder.toString();
     }
+    
+    /**
+     * Returns a concatenated version of the printable strings of each object.
+     */
+   static String getPrintableString(Printable... printables){
+	   final StringBuilder printablesBuilder = new StringBuilder();
+	   for(Printable item: printables) {
+		   if (item == null) {
+			   continue;
+		   }
+		   printablesBuilder.append(item.getPrintableString()).append(", ");
```

Comment:
Doesn't this result in an extra `,` at the end? ou could have used `StringJoiner` or `String.join` instead.